### PR TITLE
Dev Loop Snake

### PR DESCRIPTION
**Pull request by JNotelddim**
Just adding the ability for the snake to follow its own tail and stuff. 
Am now realizing I didn't write tests to go along with it :v: woops :P

### CODE REVIEW

File: src/brain.py [hunk 1]
```diff
@@ -86,3 +93,144 @@ def get_moves_for_nearest_food(self) -> Optional[List[str]]:
             return None
 
         return options
+
+    def follow_tail(self) -> List[str]:
+        """Get moves for getting to tail."""
+        tail = self.me.coordinates[len(self.me.coordinates) - 1]
```
Comment (mattcasiro) [inline src/brain.py]: - [x] `self.me.coordinates[-1]`

File: src/brain.py [hunk 2]
```diff
@@ -20,10 +22,15 @@ def get_decision(self) -> str:
         valid_moves = self.get_valid_moves()
 
         if not valid_moves:
-            return "left"
```
Comment (mattcasiro) [inline src/brain.py:23]: Can we comment the logic here of why we're handling 'there's no valid move' in this way?

- [x] Add comment

File: app/api.py [hunk 3]
```diff
@@ -21,8 +21,8 @@ def start_response(color):
     )
 
 def move_response(move):
-    assert move in ['up', 'down', 'left', 'right'], \
-        "Move must be one of [up, down, left, right]"
+#    assert move in ['up', 'down', 'left', 'right'], \
```
Comment (mattcasiro) [inline app/api.py]: Are we sure about getting rid of this? Is this for better passthrough of errors to the unit test output? Maybe we should comment the purpose of removing it so we remember the reasoning.

File: src/brain.py [hunk 4]
```diff
@@ -47,7 +54,7 @@ def get_valid_moves(self) -> List[str]:
 
     def get_nearest_food(self) -> Optional[Coordinate]:
         """Get the food item which has coordinates closest to this snake's head."""
-        closest_food = (Coordinate((0,0)), 9999.0)
+        closest_food = (Coordinate((0,0), None), 9999.0)
```
Comment (mattcasiro) [inline src/brain.py]: I think this will need to be reverted based on the changes to `Coordinate` in the other PR

- [x] Remove `None`

**Review by mattcasiro — CHANGES_REQUESTED** [review]
Love the way you set up the movement algorithms!

File: app/api.py (hunk 3 above)
Comment (JNotelddim) [reply app/api.py]: err idk that was you on my computer /shrug

**Review by mattcasiro — APPROVED** [review]
Looks good!

**Pull request merged**